### PR TITLE
[OSDEV-859] Database anonymizer schedueled task

### PR DESCRIPTION
**Pull request by roninzp**
Creates task-definitation for scheduled thas that 
   * creates temporary postgestdb instance in test AWS account from latest production snaphsot
   * run anonymization query
   * saves anonymized snapshot and removes instance

### CODE REVIEW

**Review by killev — CHANGES_REQUESTED** [review]
I don't see a task definition source. So what is executed in reality? can we add something like "echo `bla-bla`" so that we there's a way to see the result of execution

File: deployment/terraform/anonymize_db_job.tf [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "database_anonymizer_worker" {
+  statement {
+    actions = [
+      "dynamodb:GetItem",
```
[the remark this answers is not in the document]
Comment (roninzp) [reply deployment/terraform/anonymize_db_job.tf:15]: The work is not finished yet. After completion, only the necessary permissions will be here

File: deployment/terraform/anonymize_db_job.tf [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+# AWS ECS Fargate Task Definition Terraform Module
+# https://registry.terraform.io/modules/cn-terraform/ecs-fargate-task-definition/aws/latest
+module "database_anonymizer_task_definition" {
+  source                 = "git@github.com:cn-terraform/terraform-aws-ecs-fargate-task-definition.git?ref=1.0.36"
+  name_prefix            = "database-anonymizer"
+  container_image        = local.database_anonymizer_image
+  container_name         = "database-anonymizer"
+  #ephemeral_storage_size = 25
```
Comment (eugen4ernov) [inline deployment/terraform/anonymize_db_job.tf:74]: Remove the unnecessary comments

File: deployment/terraform/anonymize_db_job.tf [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      value: var.anonymized_db_s3_storage
+    }
+  ]
+
+#  secrets = [
+#    {
+#       valueFrom: "arn:aws:ssm:eu-west-1:343975343274:parameter/test",
+#       name: "TEST"
+#    },
+#  ]
```
Comment (eugen4ernov) [inline deployment/terraform/anonymize_db_job.tf:124]: rm please

File: deployment/terraform/anonymize_db_job.tf [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  event_rule_schedule_expression              = "cron(50 15 * * ? *)"
+  ecs_cluster_arn                             = module.database_anonymizer_cluster.aws_ecs_cluster_cluster_arn
+  event_target_ecs_target_subnets             = module.vpc.public_subnet_ids
+  event_target_ecs_target_task_definition_arn = module.database_anonymizer_task_definition.aws_ecs_task_definition_td_arn
+  ecs_execution_task_role_arn                 = aws_iam_role.database_anonymizer.arn
+  ecs_task_role_arn                           = aws_iam_role.database_anonymizer.arn
+}
+
+module "ecr_repository_database_anonymizer" {
+  source = "github.com/azavea/terraform-aws-ecr-repository?ref=1.0.0"
```
Comment (eugen4ernov) [inline deployment/terraform/anonymize_db_job.tf:151]: Consider hosting the modules by yourselves

File: deployment/terraform/anonymize_db_job.tf [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+  repository_name =  "${lower(replace(var.project, " ", ""))}-database-anonymizer-${lower(var.environment)}"
```
Comment (eugen4ernov) [inline deployment/terraform/anonymize_db_job.tf:153]: You can set these constructions in files like local.tf and then use local.vars here in a simpler view

File: deployment/terraform/anonymize_db_job.tf [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+      value: "anondb"
+    },
+    {
+      name: "POSTGRES_PASSWORD",
+      value: "anondb"
+    },
+    {
+      name: "POSTGRES_DB",
+      value: "anon"
+    },
```
Comment (eugen4ernov) [inline deployment/terraform/anonymize_db_job.tf:88]: Are these values real?

Comment (roninzp) [reply deployment/terraform/anonymize_db_job.tf:88]: This temporary database is recreated each time the scheduled task runs.

File: deployment/terraform/config.tf [hunk 7]
```diff
@@ -1,5 +1,8 @@
 provider "aws" {
   region = var.aws_region
+  default_tags {
```
Comment (killev) [inline deployment/terraform/config.tf:3]: Tags doesn't seem related this anonymisation task. Needs to be pulled out to separate PR

**Review by killev — DISMISSED** [review]

**Review by killev — APPROVED** [review]
LGTM, Approved

**Review by killev — DISMISSED** [review]

**Review by roman-stolar — APPROVED** [review]
LGTM, waiting for second approve from @eugen4ernov

**Review by killev — APPROVED** [review]
